### PR TITLE
[REF] web: `FormView`, review and simplify scss

### PR DESCRIPTION
**Pull request by Brieuc-brd**
This PR Review, simplify SCSS and adapt legacy code.

Part of the overall v16 SCSS optimization/restyle, task-2704984

Requires: 
- https://github.com/odoo/enterprise/pull/30421

task-2918464

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/form/form_controller.scss [hunk 1]
```diff
@@ -447,51 +288,8 @@ $o-form-label-margin-right: 0px;
...
-                        width: calc(100% - 24px);
-                    }
-                }
-            }
-        }
-
-        .o_form_label {
-            font-weight: $font-weight-normal;
-        }
+        --columns: 7;
```
Comment (stefanorigano) [inline addons/web/static/src/views/form/form_controller.scss:291]: In these circumstances we should probably create a generic Odoo variable and assign it here.
Eg
// Primary variables:
$form-view-group-grid-size: 7;

// Form View
--formView-group-grid: var(#{$form-view-group-grid-size}):
[...]
--columns: var(--formView-group-grid);

**Review by stefanorigano — COMMENTED** [review]
Hello @Brieuc-brd ,  thanks.
some preliminary remarks just looking at the code

File: addons/web/static/src/views/form/form_controller.scss (hunk 1 above)
Comment (Brieuc-brd) [reply addons/web/static/src/views/form/form_controller.scss]: To solve a layout issue, I had to use the classic BS flex grid for the `o_group` element.
So this variable is not necessary anymore.

**Comment by Brieuc-brd** [comment]
Hello @stefanorigano @adr-odoo 👋
This PR is ready for review. Could you take a look at it ?

File: addons/web/static/src/views/form/form_group/form_group.xml [hunk 2]
```diff
@@ -2,44 +2,42 @@
...
-      <tbody>
-          <td t-if="props.slots.title" t-attf-colspan="{{ props.maxCols }}" style="width: 100%">
+    <div t-attf-class="{{ allClasses }} {{ props.class and props.class.indexOf('oe_subtotal_footer') >=0 ? 'd-grid gap-0 text-end' : 'd-sm-grid' }}" class="o_inner_group" t-if="props.slots" t-att-style="props.style">
+        <div t-if="props.slots.title" t-attf-style="grid-column: span {{ props.maxCols }}">
               <t t-slot="title" />
-          </td>
-          <tr t-foreach="getRows()" t-as="row" t-key="row_index">
-            <t t-foreach="row" t-as="cell" t-key="cell_index" >
+        </div>
+        <div t-foreach="getRows()" t-as="row" t-key="row_index" class="o_wrap_field mb-3 mb-sm-0" t-attf-class="{{ props.class and props.class.indexOf('oe_subtotal_footer') >=0 ? 'd-contents' : 'd-flex d-sm-contents flex-column' }}" t-if="row.isVisible">
```
Comment (stefanorigano) [inline addons/web/static/src/views/form/form_group/form_group.xml:19]: Not sure if it's the line causing the issue, but on mobile the (visual) distance between fields is lower that the distance between labels and inputs.. it makes very difficult to scan.
It was the case already before, but here seems a bit worst 

image

**Review by stefanorigano — CHANGES_REQUESTED** [review]
Hello @Brieuc-brd , some remarks

File: addons/web/static/src/views/form/form_controller.scss [hunk 3]
```diff
@@ -915,48 +673,43 @@ $o-form-label-margin-right: 0px;
...
-        > tr {
-            display: flex;
-            flex-flow: row wrap;
-
-            > td {
-                flex: 1 0 auto;
-                display: block;
-                max-width: 100%;
-                padding: 0;
+    .o_wrap_field {
```
[the remark this answers is not in the document]
Comment (Brieuc-brd) [reply addons/web/static/src/views/form/form_controller.scss:679]: Fixed. Although we have changed the class name for the label container (`o_td_label` ➡️ `o_wrap_label`), we still need to consider the old name. This condition could be removed when all app will be revamped :
https://github.com/odoo/odoo/pull/98101/files#diff-26a45bff860d45f36003c55996b92ced0762c62dcd94d5d79bf7b5e4094a1ee0R312

File: addons/web/static/src/views/form/form_group/form_group.xml (hunk 2 above)
Comment (Brieuc-brd) [reply addons/web/static/src/views/form/form_group/form_group.xml:19]: Fixed. I put the style of label on mobile back and adapted the style of `o_horizontal_separator` to increase the readability.

File: addons/web/static/src/views/form/button_box/button_box.xml [hunk 4]
```diff
@@ -5,13 +5,12 @@
...
     <t t-set="visibleButtons" t-value="allButtons['visible']" />
     <t t-set="additionalButtons" t-value="allButtons['additional']" />
-    <div class="o-form-buttonbox oe_button_box" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
-        <t t-slot="{{ button_value }}" t-foreach="visibleButtons" t-as="button" t-key="button_value" />
-
+    <div class="o-form-buttonbox oe_button_box position-relative d-block text-end" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
+        <t t-slot="{{ button_value }}" t-foreach="visibleButtons" t-as="button" t-key="button_value"/>
         <t t-if="additionalButtons.length" >
-            <Dropdown togglerClass="'btn o_button_more'" class="'btn oe_stat_button'">
+            <Dropdown togglerClass="'btn btn-light o_button_more border-0 border-start border-bottom'" class="'btn oe_stat_button position-relative p-0 border-0'" menuClass="'position-absolute start-0 top-100 w-100 p-0 border-0'">
```
[the remark this answers is not in the document]
Comment (Brieuc-brd) [reply addons/web/static/src/views/form/button_box/button_box.xml:11]: Fixed. I set `--o-input-border-color` to `$gray-600`

**Comment by Brieuc-brd** [comment]
Hey @stefanorigano 👋 
Thanks for your feedback ! I updated the PR 👍

**Comment by Brieuc-brd** [comment]
Since we can't reproduce all the cases of this view with `grid` for now, this PR reviews the css keeping the `table` structure.

cc. @stefanorigano

File: addons/web/static/src/views/form/button_box/button_box.xml [hunk 5]
```diff
@@ -5,13 +5,12 @@
     <t t-set="allButtons" t-value="getButtons()" />
     <t t-set="visibleButtons" t-value="allButtons['visible']" />
     <t t-set="additionalButtons" t-value="allButtons['additional']" />
-    <div class="o-form-buttonbox oe_button_box" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
-        <t t-slot="{{ button_value }}" t-foreach="visibleButtons" t-as="button" t-key="button_value" />
-
+    <div class="o-form-buttonbox oe_button_box position-relative d-block text-end" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
```
Comment (adr-odoo) [inline addons/web/static/src/views/form/button_box/button_box.xml:8]: d-block already here, needed on o-form-buttonbox too? also check if really needed as div is supposed to be a block by default

Comment (Brieuc-brd) [reply addons/web/static/src/views/form/button_box/button_box.xml]: Indeed, it was not necessary. I removed it 👍

File: addons/web/static/src/views/form/form.variables.scss [hunk 6]
```diff
@@ -0,0 +1 @@
+$o-form-view-sheet-max-width: 1141px !default;
```
Comment (adr-odoo) [inline addons/web/static/src/views/form/form.variables.scss:1]: why 1 px more than before?

File: addons/web/static/src/views/form/button_box/button_box.xml [hunk 7]
```diff
@@ -5,13 +5,12 @@
     <t t-set="allButtons" t-value="getButtons()" />
     <t t-set="visibleButtons" t-value="allButtons['visible']" />
     <t t-set="additionalButtons" t-value="allButtons['additional']" />
-    <div class="o-form-buttonbox oe_button_box" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
-        <t t-slot="{{ button_value }}" t-foreach="visibleButtons" t-as="button" t-key="button_value" />
-
+    <div class="o-form-buttonbox oe_button_box position-relative text-end" t-attf-class="{{ visibleButtons.length >= getMaxButtons() ? 'o_full' : 'o_not_full'}} {{this.props.class}}">
```
Comment (adr-odoo) [inline addons/web/static/src/views/form/button_box/button_box.xml:8]: position-relative already set in o-form-buttonbox

**Review by adr-odoo — APPROVED** [review]

**Pull request closed without merging**